src/fermilib/transforms/_test_bksf.py

```diff
@@ -0,0 +1,176 @@
+        correct_operators_a13 = ((0, 'Z'), (2, 'Z'), (3, 'Z'),(4,'X'))
+        correct_operators_a23 = ((1,'Z'), (2, 'Z'), (3,'Z'),(4, 'Z'), (5, 'X'))
+        
+        qterm_a01 = QubitOperator(correct_operators_a01, 1)
+        qterm_a02 = QubitOperator(correct_operators_a02, 1)
+        qterm_a03 = QubitOperator(correct_operators_a03, 1)
+        qterm_a12 = QubitOperator(correct_operators_a12, 1)
+        qterm_a13 = QubitOperator(correct_operators_a13, 1)
+        qterm_a23 = QubitOperator(correct_operators_a23, 1)
+        
+        
+        self.assertTrue(qterm_a01.isclose(_bksf.bksf_edge_operator_aij(edge_matrix,0,1)))
+        self.assertTrue(qterm_a02.isclose(_bksf.bksf_edge_operator_aij(edge_matrix,0,2)))
+        self.assertTrue(qterm_a03.isclose(_bksf.bksf_edge_operator_aij(edge_matrix,0,3)))
+        self.assertTrue(qterm_a12.isclose(_bksf.bksf_edge_operator_aij(edge_matrix,1,2)))                
+        self.assertTrue(qterm_a13.isclose(_bksf.bksf_edge_operator_aij(edge_matrix,1,3)))
+        self.assertTrue(qterm_a23.isclose(_bksf.bksf_edge_operator_aij(edge_matrix,2,3)))
+
+
+    def test_bksf_jw_number_operator(self):
+        bksf_n = bksf_number_operator(self.molecular_hamiltonian)
+        from fermilib.utils import count_qubits
+        jw_n=QubitOperator()
+        for i in range(count_qubits(self.molecular_hamiltonian)):
+            jw_n+=jordan_wigner_one_body(i,i)
+        jw_eig_spec= eigenspectrum(jw_n)
+        bksf_eig_spec=eigenspectrum(bksf_n)
+        """for i in range(np.size(bksf_eig_spec)):
+            if i<np.size(jw_eig_spec):
+                print("BK    "+str(bksf_eig_spec[i])+"  JW    "+str(jw_eig_spec[i]))
+            else:
+                print("BK    "+str(bksf_eig_spec[i]))"""
+        evensector=0
+        for i in range(np.size(jw_eig_spec)):
+            if bool(np.size(np.where(jw_eig_spec[i]==bksf_eig_spec))):
+                evensector+=1
+                
+        #print("The number of eigenvalues of the Number Operator in Jordan-Wigner that are found in BKSF are "+str(evensector))
+        
+    def test_bksf_jw_hamiltonian(self):
+        bksf_H = bksf(self.molecular_hamiltonian)
+        jw_H = jordan_wigner(self.molecular_hamiltonian)
+        bksf_H_eig=eigenspectrum(bksf_H)
+        jw_H_eig=eigenspectrum(jw_H)
+        bksf_H_eig=bksf_H_eig.round(5)
+        jw_H_eig=jw_H_eig.round(5)
+        """for i in range(np.size(bksf_H_eig)):
+            if i<np.size(jw_H_eig):
+                print("BK    "+str(bksf_H_eig[i])+"  JW    "+str(jw_H_eig[i]))
+            else:
+                print("BK    "+str(bksf_H_eig[i]))"""
+        evensector=0
+        for i in range(np.size(jw_H_eig)):
+            if bool(np.size(np.where(jw_H_eig[i]==bksf_H_eig))):
+                evensector+=1
+                
+        #print("The number of eigenvalues of the Hamiltonian in Jordan-Wigner that are found in BKSF are "+str(evensector))
```

Comment:
Definitely delete commented out lines like this! The only thing that should be commented is helpful descriptions of the code.